packages/aws-cdk-lib/aws-iam/lib/private/imported-role.ts

```diff
@@ -74,8 +74,15 @@ export class ImportedRole extends Resource implements IRole, IComparablePrincipa
     }
   }
 
-  public addManagedPolicy(policy: IManagedPolicy): void {
-    Annotations.of(this).addWarningV2('@aws-cdk/aws-iam:importedRoleManagedPolicyNotAdded', `Not adding managed policy: ${policy.managedPolicyArn} to imported role: ${this.roleName}`);
+  public addManagedPolicy(policy: ManagedPolicy): void {
+    try {
+      policy.attachToRole(this);
+    } catch (e) {
+      if (e instanceof Error && e.message === 'policy.attachToRole is not a function') {
+        throw new Error('Can\'t Combine IRole with IManagedPolicy. use ManagedPolicy directly.');
```

Comment:
Throwing an error from this function is also a breaking change. Can we just add a warning as the previous behavior?

Reply:
Done.